### PR TITLE
truncate snapshot description by default, add --full option

### PR DESCRIPTION
**Pull request by kmotavalli**
I tried to implement what required by @NicoHood in #268 
Beware, I have not tested nor tried to compile the changes!

### CODE REVIEW

File: client/snapper.cc [hunk 1]
```diff
@@ -371,6 +372,7 @@ command_list(ProxySnappers* snappers, ProxySnapper*)
     const struct option options[] = {
 	{ "type",		required_argument,	0,	't' },
 	{ "all-configs",	no_argument,		0,	'a' },
+	{ "full",	no_argument,	0, 'f' },
```
Comment (aschnell) [inline client/snapper.cc]: "full" is a bit ambiguous for my taste. Please use ''no-abbrev" like zypper does.

File: client/snapper.cc [hunk 2]
```diff
@@ -380,6 +382,12 @@ command_list(ProxySnappers* snappers, ProxySnapper*)
 	cerr << _("Command 'list' does not take arguments.") << endl;
 	exit(EXIT_FAILURE);
     }
+	
+	bool full = false;
+	if ((opt = opts.find("full")) != opts.end())
```
Comment (aschnell) [inline client/snapper.cc]: opt is not know here. Move the lines a bit lower.

File: client/snapper.cc [hunk 3]
```diff
@@ -450,6 +458,12 @@ list_from_one_config(ProxySnapper* snapper, ListMode list_mode)
 	    header.add(_("Description"));
 	    header.add(_("Userdata"));
 	    table.setHeader(header);
+	    
+	    //truncate description?
+	    if (full == false)
```
Comment (aschnell) [inline client/snapper.cc]: Use ```!full```.

**Comment by aschnell** [comment]
I have commented some minor issues. But for whatever reason it just does not work for me.

**Comment by kmotavalli** [comment]
do column numbers start at 0? I may be trying to set the property on the wrong column. I'll try to compile and test what's wrong this evening

**Comment by kmotavalli** [comment]
i changed column numbers. it seems to work as intended now in gdb:

(gdb) run list --no-abbrev
`pre    | 210 |       | Thu 12 Jan 2017 01:10:53 PM CET | root | number  | zypp(zypper) in modem-manager-gui                             | important=no
post   | 211 | 210   | Thu 12 Jan 2017 01:11:10 PM CET | root | number  |                                                               | important=no
pre    | 212 |       | Thu 12 Jan 2017 03:40:42 PM CET | root | number  | zypp(zypper) in boost-devel libacl-devel libxml2-devel libb-> | important=no
pre    | 213 |       | Thu 12 Jan 2017 03:41:48 PM CET | root | number  | zypp(zypper) in boost-devel libacl-devel libxml2-devel libb-> | important=no
post   | 214 | 213   | Thu 12 Jan 2017 03:41:54 PM CET | root | number  |                                                               | important=no
`

lines are not wrapped.

(gdb) run list --no-abbrev

`pre    | 210 |       | Thu 12 Jan 2017 01:10:53 PM CET | root | number  | zypp(zypper) in modem-manager-gui                                                                                          | important=no
post   | 211 | 210   | Thu 12 Jan 2017 01:11:10 PM CET | root | number  |                                                                                                                            | important=no
pre    | 212 |       | Thu 12 Jan 2017 03:40:42 PM CET | root | number  | zypp(zypper) in boost-devel libacl-devel libxml2-devel libbtrfs-devel libmount-devel dbus-1-devel docbook-xsl-stylesheets  | important=no
pre    | 213 |       | Thu 12 Jan 2017 03:41:48 PM CET | root | number  | zypp(zypper) in boost-devel libacl-devel libxml2-devel libbtrfs-devel libmount-devel dbus-1-devel docbook-xsl-stylesheets  | important=no
`

full output is shown with --no-abbrev.

it doesn't seem to work in a normal terminal window

**Comment by kmotavalli** [comment]
i think this statement should be removed: https://github.com/openSUSE/snapper/blob/master/client/utils/console.cc#L22

**Pull request closed without merging**